tests/python/test_input_checker.py

```diff
@@ -0,0 +1,321 @@
+
+    @staticmethod
+    def get_empty_file(tmp_path):
+        profile_file = tmp_path / "file"
+        profile_file.touch()
+
+        return profile_file
+
+
+class TestCheckCandidateOptionType:
+
+    def test_fail_if_no_existing_option_is_given(self):
+        option = "ThisCantBeAnOption"
+        value = "notAValue"
+
+        with pytest.raises(UnrecognizedCandidateOptionType):
+            check_candidate_option_type(option, value)
+
+    def test_fail_if_max_units_value_is_negative(self):
+
+        option = "max-units"
+        value = -1545
+
+        assert check_candidate_option_type(option, value) == False
+
+    def test_fail_if_max_units_value_is_str(self):
+
+        option = "max-units"
+        value = "str"
+
+        assert check_candidate_option_type(option, value) == False
+
+    def test_true_for_obsolete_value(self):
+
+        option = "has-link-profile"
+        value = "str"
+
+        assert check_candidate_option_type(option, value) == True
+
+    def test_true_for_string_option_type(self):
+
+        option = "link"
+        value = "0-1"
+
+        assert check_candidate_option_type(option, value) == True
+
+
+class TestcheckCandidateOptionValue:
+
+    def test_fail_if_relaxed_is_not_str(self):
+        option = "relaxed"
+        value = 22
+
+        with pytest.raises(IllegalCandidateOptionValue):
+            check_candidate_option_value(option, value)
+
+    def test_accepted_relaxed_value(self):
+        option = "relaxed"
+
+        assert check_candidate_option_value(option, "true") == True
+        assert check_candidate_option_value(option, "True") == True
+        assert check_candidate_option_value(option, "FalSe") == True
+        assert check_candidate_option_value(option, "FALSE") == True
+
+
+class TestCheckCandidateName:
+
+    def test_empty_candidate_name(self):
+
+        with pytest.raises(EmptyCandidateName):
+            check_candidate_name("", "section1236")
+
+    def test_illegal_chars_in_candidate_name(self):
+
+        with pytest.raises(IllegalCharsInCandidateName):
+            check_candidate_name("newline\n123", "section1236")
+
+            check_candidate_name("escapeR\r123", "section1236")
+
+            check_candidate_name("tab\t123", "section1236")
+
+            check_candidate_name("escapeF\f\v123", "section1236")
+
+            check_candidate_name("escapeV\v123", "section1236")
+
+            check_candidate_name("parenthesis(123", "section1236")
+
+
+class TestCheckCandidateLink:
+
+    def test_empty_link(self):
+
+        with pytest.raises(EmptyCandidateLink):
+            check_candidate_link("", "section0")
+
+    def test_fail_if_link_doesnt_contains_separator(self):
+
+        with pytest.raises(CandidateLinkWithoutSeparator):
+
+            check_candidate_link("atob", "section")
+            check_candidate_link("ato-b", "section")
+            check_candidate_link("ato- b", "section")
+            check_candidate_link("ato -b", "section")
```

Comment:
same here,

with pytest.raises(CandidateLinkWithoutSeparator)

should be used for each check